sdc/rabbit/consumer.py

```diff
@@ -0,0 +1,149 @@
+import logging
+
+from structlog import wrap_logger
+
+from sdc.rabbit.exceptions import BadMessageError, DecryptError, RetryableError
+from sdc.rabbit.publisher import QueuePublisher
+
+logger = wrap_logger(logging.getLogger('__name__'))
+
+
+class MessageConsumer():
+    """This is a queue consumer that handles messages from RabbitMQ message queues.
+
+    On receipt of a message it takes a number of params from the message
+    properties, processes the message, and (if successful) positively
+    acknowledges the publishing queue.
+
+    If a message is not successfuly processed, it can be either negatively
+    acknowledged, rejected or quarantined, depending on the type of excpetion
+    raised.
+
+    """
+    @staticmethod
+    def delivery_count(properties):
+        """
+        Returns the delivery count for a message from the rabbit queue. The
+        value is auto-set by rabbitmq. Will raise KeyError if x-delivery-count
+        is missing from message headers.
+
+
+        :param properties: Message properties
+
+        :returns: Delivery count value
+        :rtype: int
+
+        """
+        delivery_count = 0
+        delivery_count = properties.headers['x-delivery-count']
+        return delivery_count + 1
+
+    @staticmethod
+    def tx_id(properties):
+        """
+        Gets the tx_id for a message from a rabbit queue, using the
+        message properties. Will raise KeyError if tx_id is missing from message
+        headers.
+
+        : param properties: Message properties
+
+        : returns: tx_id of survey response
+        : rtype: str
+        """
+        tx = properties.headers['tx_id']
+        logger.info("Retrieved tx_id from message properties: tx_id={}".format(tx))
+        return tx
+
+    def __init__(self, consumer, process):
+        """Create a new instance of the MessageConsumer class
+
+        : param consumer: Object of type sdc.rabbit.AsyncConsumer
+        : param process: Function or method to use for processsing message. Will
+            be passed the body of the message as a string decoded using UTF - 8.
+            Should raise sdc.rabbit.DecryptError, sdc.rabbit.BadMessageError or
+            sdc.rabbit.RetryableError on failure, depending on the failure mode.
+
+        :returns: Object of type Consumer
+        :rtype: Consumer
+
+        """
+        self._consumer = consumer
+        self._process = process
+
+        if not callable(process):
+            msg = 'callback is not callable'
+            raise AttributeError(msg.format(process))
+        self._quarantine_publisher = QueuePublisher(self._consumer._rabbit_urls,
+                                                    self._consumer._rabbit_quarantine_queue)
```

Comment:
What if someone changes it in the future without realising their accessed from outside the class